Lynk/frontend/src/components/contacts-table-components/contacts-table.jsx

```diff
@@ -24,113 +24,39 @@ const ContactsTable = () => {
-        return;
-      }
-
-      const { column, condition, value } = filter;
-      const dbColumn = getDbColumnName(column);
-
-      const arrayColumns = ['relationship_type', 'tags'];
-      const isArrayColumn = arrayColumns.includes(column);
-
-      switch (condition) {
-        case 'contains':
-          if (isArrayColumn) {
-            filteredQuery = filteredQuery.cs(dbColumn, [value]);
-          } else {
-            filteredQuery = filteredQuery.ilike(dbColumn, `%${value}%`);
-          }
-          break;
-        case 'does_not_contain':
-          if (isArrayColumn) {
-            filteredQuery = filteredQuery.not(dbColumn, 'cs', [value]);
-          } else {
-            filteredQuery = filteredQuery.not(dbColumn, 'ilike', `%${value}%`);
-          }
-          break;
-        case 'is':
-          if (isArrayColumn) {
-            filteredQuery = filteredQuery.eq(dbColumn, [value]);
-          } else {
-            filteredQuery = filteredQuery.eq(dbColumn, value);
-          }
-          break;
-        case 'is_not':
-          if (isArrayColumn) {
-            filteredQuery = filteredQuery.neq(dbColumn, [value]);
-          } else {
-            filteredQuery = filteredQuery.neq(dbColumn, value);
-          }
-          break;
-        case 'is_empty':
-          if (isArrayColumn) {
-            filteredQuery = filteredQuery.or(`${dbColumn}.is.null,${dbColumn}.eq.{}`);
-          } else {
-            filteredQuery = filteredQuery.or(`${dbColumn}.is.null,${dbColumn}.eq.`);
-          }
-          break;
-        case 'is_not_empty':
-          if (isArrayColumn) {
-            filteredQuery = filteredQuery.not(dbColumn, 'is', null).neq(dbColumn, '{}');
-          } else {
-            filteredQuery = filteredQuery.not(dbColumn, 'is', null).neq(dbColumn, '');
-          }
-          break;
-        default:
-          console.warn(`Unknown filter condition: ${condition}`);
-          break;
-      }
-    });
-
-    return filteredQuery;
-  };
 
 
   useEffect(() => {
 
     if (!session) return;
 
-    const fetchContacts = async () => {
-      let query = supabase
-        .from("connections").select("*").eq("user_id", session?.user?.id);
+    const fetchContactsData = async () => {
+      const result = await fetchContacts(session?.user?.id, filters);
 
-      query = applyFiltersToQuery(query, filters);
-      const { data, error } = await query;
-
-      if (error) {
-        console.error(error);
+      if (!result.success) {
+        console.error(result.error);
         toast.error("There was an error fetching the contacts");
       } else {
-        setData(data);
-        console.log(data);
+        setData(result.data);
+        console.log(result.data);
       }
     };
 
-    fetchContacts();
+    fetchContactsData();
   }, [session, filters])
 
-  const onDeleteContact = (id) => {
+  const onDeleteContact = async (id) => {
     console.log(id)
```

Comment:
Do you intend to keep this console log as a permanent feature? This is used for engineering testing, but generally in many places, it is a better idea to log information you need another way. But if this is all you have for now, that's ok!